querybook/server/lib/github_integration/github_integration.py

```diff
@@ -0,0 +1,110 @@
+import certifi
+from flask import session as flask_session, request
+from app.auth.github_auth import GitHubLoginManager
+from env import QuerybookSettings
+from lib.logger import get_logger
+from app.flask_app import flask_app
+from typing import Optional, Dict, Any
+
+LOG = get_logger(__file__)
+
+
+GITHUB_OAUTH_CALLBACK = "/github/oauth2callback"
+
+
+class GitHubIntegrationManager(GitHubLoginManager):
+    def __init__(self, additional_scopes: Optional[list] = None):
+        self.additional_scopes = additional_scopes or []
+        super().__init__()
+
+    @property
+    def oauth_config(self) -> Dict[str, Any]:
+        config = super().oauth_config
+        config["scope"] = "user email " + " ".join(self.additional_scopes)
+        config[
+            "callback_url"
+        ] = f"{QuerybookSettings.PUBLIC_URL}{GITHUB_OAUTH_CALLBACK}"
+        return config
+
+    def save_github_token(self, token: str) -> None:
+        flask_session["github_access_token"] = token
```

Comment:
let's define a const for the key `github_access_token` as it will be used by other module